src/Properties.ts

```diff
@@ -0,0 +1,28 @@
+import { Logger } from '@hmcts/nodejs-logging'
+import * as fs from 'fs'
+
+const logger = Logger.getLogger('applicationRunner')
+
+export function addTo (config: any, mountPoint: fs.PathLike = '/mnt/secrets/', propertiesPrefix: string = 'keyVault') {
+  logger.info(`Reading properties from volume: '${mountPoint}'`)
+  config[propertiesPrefix] = fs.readdirSync(mountPoint, null)
+    .reduce((obj, dir) => addDir(dir, obj, mountPoint), {})
+}
+
+function addDir (dir: string, obj: any, mountPoint: fs.PathLike): any {
+  obj[dir] = readDirectories(mountPoint, dir).reduce((values, file) => addFile(values, file, mountPoint, dir), {})
+  return obj
+}
+
+function addFile (values: any, file: string, mountPoint: fs.PathLike, dir: string) {
+  values[file] = readFile(mountPoint, dir, file).trim()
+  return values
+}
+
+function readFile (mountPoint: fs.PathLike, dir: string, file: string) {
+  return fs.readFileSync(mountPoint + '/' + dir + '/' + file, 'utf8')
+}
+
+function readDirectories (mountPoint: fs.PathLike, dir: string) {
+  return fs.readdirSync(mountPoint + '/' + dir)
```

Comment:
Hmm what do you link about reading the directories asynchronously?
It's pretty bad practice in node to call the sync methods

Reply:
yup ... did try use the async fs promises but this means we have to start express in an async manor so that the properties are loaded by the time it starts. 

I thought this would be too of a large change to the application structure of every application to be worth the minor benefit of async.

Reply:
can't you await the promises being completed before continuing?

`await addTo(config)`

That way all directories can be loaded async

Reply:
does it really matter?
I tried this earlier but it was not working as expected since something to do with await has to be async function, or something like that so I moved onwards.

Reply:
you need to add the `async` keyword to any function that is called in an async chain